### PR TITLE
Modification of abi format to include gas cost estimate for functions…

### PR DESCRIPTION
**Pull request by fubuloubu**
… (skips constructor)

### CODE REVIEW

File: viper/compiler.py [hunk 1]
```diff
@@ -32,6 +27,19 @@ def gas_estimate(origcode, *args, **kwargs):
     return o
 
 
+def mk_full_signature(code, *args, **kwargs):
+    abi = parser.mk_full_signature(parser.parse(code))
+    # Add gas estimates for each function to ABI
+    gas_estimates = gas_estimate(code)
+    for idx, func in enumerate(abi):
+        func_name = func['name'].split('(')[0]
+        print(func_name)
```
Comment (fubuloubu) [inline viper/compiler.py]: Duh, fixed. Thanks!

**Comment by fubuloubu** [comment]
Had to reorder the commits because I tried re-basing this branch and made a mistake.

**Comment by fubuloubu** [comment]
@vbuterin see any problems with this?

**Comment by fubuloubu** [comment]
@DavidKnott What do you think of this update? Is there a potential problem with adding a field to the ABI output?

**Comment by DavidKnott** [comment]
I don't think so, I'll look into it

**Comment by DavidKnott** [comment]
@fubuloubu I think as of now gas estimation isn't accurate enough to be included in the abi.  Right now gas estimation fails on quite a few the tests in `test_parser.py`.  I think gas estimation functionality needs to be improved before this gets merged.

**Comment by fubuloubu** [comment]
I think it would be easier to debug if that information was actually available in the ABI as it at least shows a bigger picture, and also if the gas estimation feature were working correctly then this update still works the same way and provides those more accurate updates in the same way. I don't believe this update creates any new bugs as it is fairly simple and unobtrusive, and #306 shows it is a useful feature to have in the long term.

This is alpha software after all, no guarantees on correct output just yet.

**Comment by DavidKnott** [comment]
@fubuloubu Looked at this again and I think you're right, merging it in.

**Pull request merged**